pkg/dogstatsd/server.go

```diff
@@ -250,9 +272,15 @@ func (s *Server) worker(metricOut chan<- *metrics.MetricSample, eventOut chan<-
 						dogstatsdMetricParseErrors.Add(1)
 						continue
 					}
+
+					if s.debugMetricsStats {
+						s.storeMetricsStats(sample.Tags)
```

Comment:
Could have used a channel there, but I still would need a lock for the map (because the call of the command dogstatsd-stats (read) could be in the same time as writing processed metrics stats (write)).
We wouldn't benefit from the internal lock of the channel, thus, the only gain we would have from using a channel is that another routine would processed the map writes. However, I didn't see slow-down (in pkt processed per seconds) after having introduced this change, I've decided to go for the simplest implementation.

Reply:
Using channels would move the lock away from the hot path in another go-routine. I think this is desirable if we care about performance here.